packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -82,7 +92,17 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
   void _handlePointerDataPacket(ui.PointerDataPacket packet) {
     // We convert pointer data to logical pixels so that e.g. the touch slop can be
     // defined in a device-independent manner.
-    _pendingPointerEvents.addAll(PointerEventConverter.expand(packet.data, window.devicePixelRatio));
+    final Iterable<PointerEvent> events =
+        PointerEventConverter.expand(packet.data, window.devicePixelRatio);
+    if (resamplingEnabled) {
```

Comment:
Correct me if there's other reason, but can we move this part to `_flushPointerEventQueue`, so 
1. `_pendingPointerEvents` will mean the same thing with or without the resampler; 
2. relevant codes will be at the same place so it's easier to understand and maintain.

Reply:
Good idea. I've done this in latest change. I've also changed to resampler code to take a callback as that integrates better gesture bindings and avoids unnecessary use of a temporary list.